packages/types/src/prop-types.ts

```diff
@@ -44,15 +48,17 @@ type DataAttr = {
   [key in `data-${string}`]?: string | number | Booleanish
 }
 
-export type PropTypes = Record<"button" | "label" | "input" | "output" | "element" | "select" | "style", Dict>
+export type PropTypes<T = any> = Record<
+  "button" | "label" | "input" | "output" | "element" | "select" | "style",
+  T extends SplitArgs ? SplitArgs : Dict
+>
```

Comment:
How about using the `T` type param and setting it to `Dict` by default?

```suggestion
export type PropTypes<T = Dict> = Record<
  "button" | "label" | "input" | "output" | "element" | "select" | "style",
  T
>
```